Code/Tools/AssetProcessor/native/AssetManager/FileStateCache.cpp

```diff
@@ -163,16 +165,22 @@ namespace AssetProcessor
 
     QString FileStateCache::PathToKey(const QString& absolutePath) const
     {
-        QString normalized = AssetUtilities::NormalizeFilePath(absolutePath);
+        auto cached = m_keyCache.find(absolutePath);
```

Comment:
I'm surprised this is faster.  Isn't find just hashing the given path in order to look it up anyway?  Assuming the keycache is an unordered set?  If its faster, its purely becuase its avoiding the UTF8 conversion.   Its frustrating becuase in the vast majority of cases we don't actually have to lowcase or anything.  Perhaps we could use a custom hash function of some sort that doesn't involve the entire string or something

Reply:
considering that m_keyCache lookup and store both depend no qhash(Qstring) which is implemented as
```
static uint hash(const QChar *p, int n)
{
    uint h = 0;
    uint g;

    while (n--) {
        h = (h << 4) + (*p++).unicode();
        if ((g = (h & 0xf0000000)) != 0)
            h ^= g >> 23;
        h &= ~g;
    }
    return h;
}
```

Reply:
perhaps its purely avoiding the tolower.

Reply:
I believe it is the tolower mostly, but the qt clean function was also showing up in the profiler (and it is actually doing something, I saw paths that included .. in them).  I didn't dig into qt code to see what its doing under the hood though.  I also tried using AZ::IO::Path but that was surprisingly more expensive to use as a key compared to QString

Reply:
Yeah I'll take a 2x speedup anyday.  For future there might be additional speedup opportunities here, by using a different hash function or something like that.